datalad/crawler/nodes/tests/test_misc.py

```diff
@@ -471,4 +473,12 @@ def node(d):
 def test_debug():
     yield _test_debug, "About to run"
     yield _test_debug, "Ran node .* which yielded 1 times", ('after',)
-    yield _test_debug, "Ran node .* which yielded 0 times", ('empty',)
\ No newline at end of file
+    yield _test_debug, "Ran node .* which yielded 0 times", ('empty',)
+
+
+def test_fix_url():
+    test_url = 'http://myweb.fsu.edu/bgomez/AJPS_2001_Erratum.zip'
+    # urlTest = 'http://myweb.fsu.edu/bgomez/GomezWilson_2006_JOP_Replication Material.zip'
+    eq_(list(fix_url({'url': test_url, 'url': test_url + ' ', 'url': test_url + '  ',
+                      'url': ' ' + test_url, 'url': '  ' + test_url})),
+        [{'url': test_url, 'url': test_url, 'url': test_url, 'url': test_url, 'url': test_url}])
```

Comment:
this is a dictionary, so 
```{'url': test_url, 'url': test_url, 'url': test_url, 'url': test_url, 'url': test_url}``` 
is identical to
```{'url': test_url}```
;)  make it into separate tests, e.g.

```python
eq_(list(fix_url({'url': "http://site/u r"})) == [{'url': "http://site/u%20r"}])
eq_(list(fix_url({'url': "http://site/ur "})) == [{'url': "http://site/ur%20"}])  # trailing spaces are still spaces
```